packages/annotations/test/annotations.test.ts

```diff
@@ -0,0 +1,63 @@
+import { getAnnotationServiceUrl, searchAnnotations } from "../src/index";
+import {
+  annoSearchResponse,
+  emptyAnnoSearchResponse
+} from "./mocks/ago_search";
+import { annoQueryResponse } from "./mocks/anno_search";
+
+import * as fetchMock from "fetch-mock";
+import { IQueryFeaturesResponse } from "@esri/arcgis-rest-feature-service";
+
+describe("getAnnotationServiceUrl", () => {
+  afterEach(fetchMock.restore);
+
+  it("should return an annotation service url", done => {
+    // i'd prefer to mock the specific request, but thats not working
+    // https://www.arcgis.com/sharing/rest/search?f=json&q=hubAnnotationLayer%20AND%20orgid%3A5bc
+    fetchMock.once("*", annoSearchResponse);
+
+    getAnnotationServiceUrl("5bc").then(response => {
+      const [url, options]: [string, RequestInit] = fetchMock.lastCall("*");
+      expect(options.method).toBe("GET");
+      expect(url).toContain(
+        "q=typekeywords%3AhubAnnotationLayer%20AND%20orgid%3A5bc"
+      );
+      expect(response).toBe(annoSearchResponse.results[0].url);
+      done();
+    });
+  });
+
+  it("should throw if no annotation layer is found", done => {
+    fetchMock.once("*", emptyAnnoSearchResponse);
+
+    getAnnotationServiceUrl("v8b").catch(error => {
+      const [url, options]: [string, RequestInit] = fetchMock.lastCall("*");
+      expect(options.method).toBe("GET");
+      expect(url).toContain(
+        "q=typekeywords%3AhubAnnotationLayer%20AND%20orgid%3Av8b"
+      );
+      expect(error).toBe(
+        "No annotation service found. Commenting is likely not enabled."
+      );
+      done();
+    });
+  });
+});
+
+describe("searchAnnotations", () => {
+  afterEach(fetchMock.restore);
+
+  it("should query for annotations when no parameters are passed", done => {
+    fetchMock.once("*", annoQueryResponse);
```

Comment:
in the long run it would probably better to stub `queryFeatures()` and verify that expected params were passed, but this'll get'r done for now

Reply:
i was already verifying the request parameters inside the `then()` block. so i _think_ you're picking on the fact that i'm mocking requests so generically (ie: just listening for `"*"`)? 

if so, we do it all over the place in `arcgis-rest-*` packages and i've never felt quite right about it, so i pushed up another commit here that sets up the mock server to be a bit more selective in the requests it will respond to.